### ISSUE TITLE
CUDA 12.6 Inductor per test failures

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#148699]
### 🐛 Describe the bug

We are working on PR to move our Inductor CUDA 12.4 -> 12.6

This is pull request:
https://github.com/pytorch/pytorch/pull/148612/

We do see 2 test failures and 1 unexpected pass

fail_accuracy:
timm_efficientnet,fail_accuracy,7
crossvit_9_240,fail_accuracy,7

pass:
tinynet_a,pass,6

Workflows:
https://github.com/pytorch/pytorch/pull/148612/commits/29ec07bdd63545ec7f38970f4e48bf72be515f8e#diff-edec9c9b528f9ee5f7113d88ce1e776897a63c5ac0260c55926a570a62a0a36b
and
https://github.com/pytorch/pytorch/actions/runs/13693492178/job/38295168222

### Versions

2.7.0 nightly

**Comment by malfet** [comment]
Removing `oncall: pt2` for now and add `high priority` to discuss if those tests are important  and if so who is on the hook to investigate what have caused the failure

**Comment by eellison** [comment]
```
E0306 09:47:52.040000 15528 site-packages/torch/_dynamo/utils.py:2878] RMSE (res-fp64): 0.00260, (ref-fp64): 0.00038 and shape=torch.Size([128]). res.dtype: torch.float32, multiplier: 3.000000, tol: 0.010000, use_larger_multiplier_for_smaller_tensor: 0
```

The errors here are not huge. Only vision models are failing - I wonder if this is maybe due to different convolution algorithms with latest CUDNN. Would it be possible to run again with `layout_opt_default=0` set here https://github.com/pytorch/pytorch/blob/e0d4c43ad104fd72fa2ae412de563e91ebfde913/torch/_inductor/config.py#L468-L470  ?

**Comment by atalman** [comment]
Hi @eellison here is the test: https://github.com/pytorch/pytorch/pull/150411

**Comment by atalman** [comment]
Moving to 2.7.1 to followup

**Comment by eqy** [comment]
Is this still visibke? We tried to repro in 12.6 internally and were not able to on A2.

**Comment by tinglvv** [comment]
crossvit_9_240 is now passing, see https://github.com/pytorch/pytorch/pull/148989
tinynet_a is passing as an improvement so no need to check. 

what is confusing is timm_efficientnet is marked here as pass in [dynamic_aot_eager_torchbench_training.csv#L63](https://github.com/pytorch/pytorch/blame/main/benchmarks/dynamo/ci_expected_accuracy/dynamic_aot_eager_torchbench_training.csv#L63) from Feb 6 commit (https://github.com/pytorch/pytorch/commit/966db82c9dbb64eac229315d785cd283dd8faec4), seems like our Mar 6 commit (https://github.com/pytorch/pytorch/commit/29ec07bdd63545ec7f38970f4e48bf72be515f8e#diff-edec9c9b528f9ee5f7113d88ce1e776897a63c5ac0260c55926a570a62a0a36b) didn't affect this file? however it correctly changed the other two files. 
Anyways, started a test PR to see if timm_efficientnet is still an issue - https://github.com/pytorch/pytorch/pull/153290

**Comment by tinglvv** [comment]
timm_efficientnet still failing with accuracy error, the difference is now much larger in test PR https://github.com/pytorch/pytorch/pull/153290. (|0.70666 - 0.21386| = 0.4928)
example failure log
```
2025-05-10T01:51:35.1542119Z cuda train timm_efficientnet 
...                 
2025-05-10T01:53:05.3984624Z E0510 01:53:05.397000 79025 site-packages/torch/_dynamo/utils.py:2948] RMSE (res-fp64): 0.70666, (ref-fp64): 0.21386 and shape=torch.Size([48, 1152, 1, 1]). res.dtype: torch.float32, multiplier: 3.000000, tol: 0.080000, use_larger_multiplier_for_smaller_tensor: 1
...
2025-05-10T01:53:05.4255800Z fail_accuracy
```

@eellison could you help take a look? Thanks.

**Comment by tinglvv** [comment]
Error seems worse with layout_opt_default=0, so I guess that doesnt work. (1.81489-0.53464=1.28025 ) https://github.com/pytorch/pytorch/actions/runs/14983900179/job/42100346931
```
2025-05-13T03:41:04.6746794Z cuda train timm_efficientnet                  
...
2025-05-13T03:42:29.9074899Z E0513 03:42:29.906000 76602 site-packages/torch/_dynamo/utils.py:2948] RMSE (res-fp64): 1.81489, (ref-fp64): 0.53464 and shape=torch.Size([10, 240, 1, 1]). res.dtype: torch.float32, multiplier: 3.000000, tol: 0.080000, use_larger_multiplier_for_smaller_tensor: 1
2025-05-13T03:42:29.9077963Z E0513 03:42:29.906000 76602 site-packages/torch/_dynamo/utils.py:2809] Accuracy failed for key name blocks.2.1.se.conv_reduce.weight.grad
...
2025-05-13T03:42:29.9349204Z fail_accuracy
```

**Comment by tinglvv** [comment]
Was also unable to reproduce this error on A10G, (failing test is on g4 like @syed-ahmed mentioned, I used very similar settings (torch==2.8.0) as in the CI. 
```
jenkins@9b1a2cebd15e:/tmp/pytorch$ python benchmarks/dynamo/torchbench.py --ci --accuracy --timing --explain --print-compilation-time --inductor --device cuda --training --amp --total-partitions 2 --partition-id 1 --only timm_efficientnet
model.safetensors: 100%|████████████████████████████████████████████████████████████████████████████████████████████| 21.4M/21.4M [00:00<00:00, 135MB/s]
loading model: 0it [00:03, ?it/s]█████████████████████████████████████████████████████████████████████████████████▎ | 21.0M/21.4M [00:00<00:00, 139MB/s]
cuda train timm_efficientnet                  
W0513 21:49:59.565000 60 site-packages/torch/_logging/_internal.py:1130] [6/0] Profiler function <class 'torch.autograd.profiler.record_function'> will be ignored
Compilation time (from dynamo_timed): 84.890943137
pass
TIMING: gc:0.00749 entire_frame_compile:69.96253 _recursive_pre_grad_passes:0.03791 pad_mm_benchmark:0.48392 _recursive_joint_graph_passes:1.04975 _recursive_post_grad_passes:0.80065 async_compile.wait:2.9644 code_gen:26.37271 inductor_compile:42.37937 backend_compile:49.90009 cudagraphify.get_container:0.45366 entire_backward_compile:14.92841 CUDAGraphNode.record:3.39791 total_wall_time:84.89094
STATS: call_* op count: 1327 | FakeTensorMode.__torch_dispatch__:57665 | FakeTensor.__torch_dispatch__:44066 | ProxyTorchDispatchMode.__torch_dispatch__:25724
Dynamo produced 3 graphs covering 1327 ops with 7 graph breaks (5 unique)
```

**Comment by atalman** [comment]
Removing from milestone, will be following up on this for release 2.8.0

**Comment by nWEIdia** [comment]
Reproduced the issue on an internal A10G. There is a ~1 out of 20 chance that the run would succeed. 
`
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
pass
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
fail_accuracy
`


Also tested with cuda 12.4, using this image: ghcr.io/pytorch/ci-image:pytorch-linux-focal-cuda12.4-cudnn9-py3-gcc9-inductor-benchmarks-e7f56771970bd2f4dac443d14937e906ea823d22

the output is below: 
`E0519 23:34:28.587000 126 site-packages/torch/_dynamo/utils.py:2890] RMSE (res-fp64): 1.49996, (ref-fp64): 0.46703 and shape=torch.Size([1152, 48, 1, 1]). res.dtype: torch.float32, multiplier: 3.000000, tol: 0.080000, use_larger_multiplier_for_smaller_tensor: 1
`

**Comment by atalman** [comment]
Removing from release 2.8 since CUDA 12.6 has Legacy status for release 2.8
